subprojects/base-services/src/main/java/org/gradle/internal/hash/HashCode.java

```diff
@@ -96,6 +96,10 @@ public int length() {
         return bytes.clone();
     }
 
+    public void appendToHasher(Hasher hasher) {
+        hasher.putBytes(bytes);
+    }
+
```

Comment:
I'd prefer if we had this in the form of `Hasher.putHashCode(HashCode)` and `MessageDigestHasher` would have the ability to access the raw `bytes` inside the `HashCode` via a package-private accessor.

Reply:
If we did that we could even merge the `Hasher` and `BuildCacheHasher` interfaces. We don't necessarily have to do that part now, though.

Reply:
fixed. I didn't unify the two interfaces yet.